content/wiki/canonical_lints.md

```diff
@@ -75,14 +75,16 @@ clippy.wildcard_dependencies = "warn"
 ## `lib.rs`
 
 ```rust
-// LINEBENDER LINT SET - lib.rs - v2
+// LINEBENDER LINT SET - lib.rs - v3
 // See https://linebender.org/wiki/canonical-lints/
-// These lints aren't included in Cargo.toml because they
-// shouldn't apply to examples and tests
-#![warn(unused_crate_dependencies)]
-#![warn(clippy::print_stdout, clippy::print_stderr)]
+// These lints aren't included in Cargo.toml because they shouldn't apply to examples and tests.
+#![cfg_attr(not(test), warn(unused_crate_dependencies))]
+#![cfg_attr(not(test), warn(clippy::print_stderr, clippy::print_stdout))]
```

Comment:
I disagree with this change; `rust-analyzer` has `cfg(test)` enabled when it runs clippy, which means that this would be unlikely to be caught locally. I also think there are two cases here:
1) Tests: Can't meaningfully warn for unused dependencies, but shouldn't really have stdout/stderr printing except for whilst debugging
2) Examples: Can't meaningfully warn for unused dependencies, but can have stdout/stderr printing

Reply:
That's fair. I don't use `rust-analyzer` for Clippy so hadn't considered it. I enabled these for `test` again.